### PR TITLE
chore: enable development mode assertions for prod_debug

### PR DESCRIPTION
**Pull request by ekashida**
## Details

Enable development mode assertions in prod_debug mode.

## Does this PR introduce breaking changes?

* ✅ `No, it does not introduce breaking changes.`

## GUS work item
W-7145172

### CODE REVIEW

File: packages/lwc/scripts/utils/rollup.js [hunk 1]
```diff
@@ -28,11 +28,15 @@ function rollupFeaturesPlugin(prod) {
...
     const compatMode = target === 'es5';
+
     return {
         inputOptions: {
             input,
             plugins: [
-                prod && rollupReplace({ 'process.env.NODE_ENV': JSON.stringify('production') }),
+                prod &&
+                    rollupReplace({
+                        'process.env.NODE_ENV': JSON.stringify(debug ? 'prod_debug' : 'production'),
```
Comment (caridy) [inline packages/lwc/scripts/utils/rollup.js]: I'm still confused about why is this here? The description of the PR doesn't tell much.

**Comment by pmdartus** [comment]
We discussed during standup and came to the conclusion that the `aura.mode=PRODDEBUG` will map to the engine `DEV` mode. @ekashida Can we close this PR?
